dev/com.ibm.ws.security.social/src/com/ibm/ws/security/social/internal/utils/OpenShiftUserApiUtils.java

```diff
@@ -244,4 +253,63 @@ JsonObject getStatusJsonObjectFromResponse(JsonObject currentResponse) throws So
+        if (response == null || response.isEmpty()) {
+            return null;
+        }
+        JsonObject jsonResponse;
+        try {
+            jsonResponse = Json.createReader(new StringReader(response)).readObject();
+        } catch (JsonParsingException e) {
+            throw new SocialLoginException("RESPONSE_NOT_JSON", null, new Object[] { response, e });
+            //RESPONSE_NOT_JSON=CWWKS53xxE: The content of the response is not a valid JSON object. The full response is {0}. {1}
+            //RESPONSE_NOT_JSON.explanation=The response is expected to be a valid JSON object.
+            //RESPONSE_NOT_JSON.useraction=Verify that request was sent to the expected target. Ensure that the intended target of the request is capable of returning JSON data. Check the response to see whether more information is included.
+        }
+        return jsonResponse;
+    }
+
+    JsonObject getJsonObjectValueFromJson(JsonObject json, String key) throws SocialLoginException {
+        if (!json.containsKey(key)) {
+            throw new SocialLoginException("JSON_MISSING_KEY", null, new Object[] { key, json });
+            //JSON_MISSING_KEY=CWWKS53xxE: The JSON object that is provided is missing an expected key: [{0}]. The full JSON object is [{1}].
+            //JSON_MISSING_KEY.explanation=The key that is specified in the message is expected to be within the JSON data. The key might be missing, or it might be in an unexpected location.
+            //JSON_MISSING_KEY.useraction=Check the JSON data to determine whether the key is missing.
+        }
+        JsonValue rawValue = json.get(key);
+        if (rawValue.getValueType() != ValueType.OBJECT) {
+            throw new SocialLoginException("JSON_ENTRY_WRONG_JSON_TYPE", null, new Object[] { key, ValueType.OBJECT, rawValue.getValueType(), json });
+            //JSON_ENTRY_WRONG_JSON_TYPE=CWWKS53xxE: The value for the key [{0}] in the JSON data is expected to be of type {1}, but the value is of type {2}. The JSON data is [{3}].
+            //JSON_ENTRY_WRONG_JSON_TYPE.explanation=The value for the specified key did not have the correct type. The JSON data might be malformed or might have an unexpected structure.
+            //JSON_ENTRY_WRONG_JSON_TYPE.useraction=Check the structure of the JSON data. Check the JSON data to see whether an error occurred.
+        }
+        return json.getJsonObject(key);
+    }
+
+    JsonObject addGroupsToResult(JsonObject rawJsonResponse, JsonObject metadataEntry) {
+        JsonObject result = metadataEntry;
+        String groupNameAttribute = config.getGroupNameAttribute();
+        if (groupNameAttribute != null && rawJsonResponse.containsKey(groupNameAttribute)) {
+            JsonObjectBuilder resultBuilder = copyJsonObject(metadataEntry);
+            resultBuilder.add(groupNameAttribute, rawJsonResponse.get(groupNameAttribute));
+            result = resultBuilder.build();
+        }
+        return result;
```

Comment:
does it matter if group is added to the end or beginning of the jsonObject object